From aad68441389eab0a7df3f0d78a3765b88388cf58 Mon Sep 17 00:00:00 2001
From: Antti Kyllonen <antti.kyllonen@consult.red>
Date: Fri, 2 Oct 2020 13:54:49 +0000
Subject: [PATCH] Configure missing dobby0 on route add

---
 src/ifvc.h |  1 +
 src/msg.c  | 13 +++++++++++++
 2 files changed, 14 insertions(+)

diff --git a/src/ifvc.h b/src/ifvc.h
index 74ba782..cfa5c44 100644
--- a/src/ifvc.h
+++ b/src/ifvc.h
@@ -27,6 +27,7 @@ struct ifmatch {
 void          iface_init              (void);
 void          iface_refresh           (void *arg);
 void          iface_exit              (void);
+int           iface_update            (int refresh);

 struct iface *iface_iterator          (int first);

diff --git a/src/msg.c b/src/msg.c
index 1dcb459..d4bc689 100644
--- a/src/msg.c
+++ b/src/msg.c
@@ -176,6 +176,19 @@ static int do_mroute4(struct ipc_msg *msg)
 			while (pos < msg->count) {
 				char *ifname_out = msg->argv[pos++];

+				// smcrouted only updates the list of interfaces on startup. However, the Dobby Networking
+				// plugin doesn't create its bridge device until the first container that needs it is
+				// launched. This means that when "smcroutectl add" is called, the Dobby bridge device is
+				// unlikely to be known to smcrouted.
+				//
+				// To fix this, we update the interface list here if dobby0 is needed and missing. Then we
+				// add the virtual interface in smcrouted, which then allows routes to be added for dobby0.
+				if (strcmp(ifname_out, "dobby0") == 0 && iface_find_by_name("dobby0") == NULL)
+				{
+					iface_update(0);
+					mroute_add_vif("dobby0", 0, DEFAULT_THRESHOLD);
+				}
+
 				iface_match_init(&state_out);
 				while ((vif = iface_match_vif_by_name(ifname_out, &state_out, NULL)) >= 0) {
 					if (vif == mroute.inbound) {
--
2.7.4